src/server.rs

```diff
@@ -554,7 +559,7 @@ impl HttpServer {
     /// # Errors
     /// `IOError` is returned when socket or epoll operations fail.
     /// `ServerFull` is returned if server full capacity has been reached.
-    fn handle_new_connection(&mut self) -> Result<()> {
+    fn handle_new_connection(&mut self, max_len: usize) -> Result<()> {
```

Comment:
You can drop this parameter from the handling function, as it's a method and you already have access to members of the struct inside.